api/schemas/Availability.yaml

```diff
@@ -0,0 +1,38 @@
+---
+type: object
+description: >
+  - If a room is not available and not booked, the room is under maintainance.
+  
+  - If a room is not available and booked, the room booked by a student.
+  
+  - If a room is available and not booked, the room available for booking.
+  
+  - If a room is available and booked, the room open for "Study With Me".
+properties:
+  id:
+    type: integer
+    format: int64
+    example: 100
+    readOnly: true
+  room-id:
+    type: integer
+    format: int64
+    example: 100
+  date:
+    type: string
+    format: date
+    example: YYYY-mm-dd
+  hours:
+    type: array
+    minItems: 0
+    maxItems: 24
+    items:
+      type: object
+      properties:
+        hour:
+          type: string
+        Available:
+          type: boolean
+        Booked:
+          type: boolean
```

Comment:
I think this is a good start. However, I'm wondering if this approach might not be feasible. Since the hours only stores up to 24 items, it seems to reflect 1 day. This would mean that we will need an `Availability` item for each day.

In our Event Storming, we talked about instead tracking `unavailabilities` (recurring and nonrecurring)._ From my understanding, this is how the general business logic will go:
* A room by default is available for booking at any time
* Admin users can dictate off-hours by creating a recurring `unavailability`. This could be an item with an time (e.g., 8:00 pm), a duration (e.g., 12 hours), and a repeat period (e.g., 24 hours).
  * Put together: this would mean that a room is not available from 8 PM to 8 AM every day
* Regular users can make individual bookings by creating a nonrecurring `unavailability`. This would likely be the same schema as a recurring one, except that it doesn't have a repeat period


I like this approach since it's easy to think about and implement and it lets us be flexible in adding new features. For example, we could explore letting students set regular meeting times by letting them create recurring `unavailabilities`

Reply:
lmk what you think of this :)

Reply:
The Availability schema is for the frontend, every request will return 0 to 24 hour objects, representing a day. It works great if we uses hour as base unit, but if we need different granularity of time, we will need a better design.